.github/workflows/appsignals-e2e-eks-test.yml

```diff
@@ -116,49 +116,51 @@ jobs:
 
       # Application pods need to be restarted for the
       # app signals instrumentation to take effect
-      - name: Restart the app pods
-        run: kubectl delete pods --all -n ${{ env.SAMPLE_APP_NAMESPACE }}
+      - name: Restart the app pods and wait for the endpoints to come online
+        id: restart-pod
+        uses: nick-fields/retry@v2
+        with:
+          max_attempts: 3
+          retry_on: error
+          timeout_minutes: 6
+          shell: bash
+          # Restart the pod, then attempt to connect to the endpoint. It will try 30 times with 10 seconds delay. If
+          # it fails to establish connection, restart the pod and try again up to 3 tries.
+          command: |
```

Comment:
I'm a little confused as to what is the root cause of the transient failure this PR is aiming to fix. Is it that the application pods are occasionally fail to run or is it the ALB endpoint that fails to come up. I feel its the latter.
In that case, does simply restarting the application pods solves the ALB endpoint issue? Have we reproduced/tested this fix anywhere?

Reply:
Yes, the failure happens when the endpoints fails to come up. We think it's happening due to occasional network problem/timeouts, so in such situations restarting the pod should hopefully allow the endpoints to come up properly. It's hard to reproduce this error, so I'm not 100% sure if it will work, but tested the code to check if the entire workflow succeeds properly and tested to see if the retry logic runs properly in the event that the commands fail.